### ISSUE TITLE
Feature Request: Customizable Hotkey for Clipboard History and Quick Pasting

### ISSUE THREAD

**Issue by VeRychard** [PasteBar/PasteBarApp#117]
PasteBar is packed with many awesome features. However, there's one crucial feature I couldn't find that is preventing me from integrating it into my everyday workflow: the ability to use customizable hotkeys for clipboard history and pins/quick pastes.

It would significantly enhance efficiency if frequently used clipboard items could be assigned to customizable hotkeys. Additionally, being able to bring up the clipboard history with a key combination would streamline the process. For example, I currently use Ditto, which greatly improves my workflow. With Ditto, I can press a button to display the clipboard history, type a few letters to search, and paste with the enter key.

Implementing this feature in PasteBar would make it much more convenient and efficient for daily use.

Thank you for considering this request, and keep up the good work!

**Comment by kurdin** [comment]
@VeRychard I agree, it would be nice to have customizable global hotkeys for clipboard history. Could you please provide a workflow and key combination you want to use to quickly paste items from clipboard history? Is something like a combination of Shift + Command (Alt) + CTRL + (1 thru 0) to paste the last copied items from 1 to 10 in your clipboard history what you have in mind?

> Additionally, being able to bring up the clipboard history with a key combination would streamline the process. For example, I currently use Ditto, which greatly improves my workflow. With Ditto, I can press a button to display the clipboard history, type a few letters to search, and paste with the enter key.

Yeah, this feature was requested before as well. I'll think about how I can implement this. Personally, I use the PasteBar menu with Recent History items to access and paste items I recently copied. You can open the menu and press keys from 1 to 0 to paste items without clicking on them.

<img width="542" alt="Screenshot 2024-07-30 at 11 55 01" src="https://github.com/user-attachments/assets/91913d0d-57b7-4547-8624-ff8072fbe9e3">

**Comment by OranShuster** [comment]
I use ditto and maccy on windows/mac but the flow is the same
WIN + V on windows/ CMD + SHIFT + V on max will show a clipboard history panel
use arrow keys to manually select, CMD/CTRL + number to paste the nth item in the list
start typing to search all the history and filter as you type

**Comment by kurdin** [comment]
> I use ditto and maccy on windows/mac but the flow is the same WIN + V on windows/ CMD + SHIFT + V on max will show a clipboard history panel use arrow keys to manually select, CMD/CTRL + number to paste the nth item in the list start typing to search all the history and filter as you type

Yeah, I will work on this when I have more time. Good feature. Thanks

**Comment by VeRychard** [comment]
Yup as @OranShuster wrote. For me I use Alt + X to bring up this menu, and then as mentioned Ctrl + numbers to paste the one you'd like to paste, or type for search.

![image](https://github.com/user-attachments/assets/42fcd1ec-8b2c-4b62-80f9-4904169b0c6b)

So basically this could be a cropped clipboard history view with the mentioned features.

@kurdin the difference from the recent history you showed is that in this popup window we can scroll through all the clipboard items, interact with them (such as special paste for texts, which is also an awesome feature, maybe I'll add another feature request for it 😄) and search in it

![image](https://github.com/user-attachments/assets/6b4dfcae-8e75-46c5-a854-955e2c436f51)

**Comment by metalelf0** [comment]
As a side note, I'd like to understand the reasoning about this app being released without this requested feature. Every clipboard manager app I've ever used (like jumpcut/flycut or maccy) is bidirectional, e.g. they seamlessly populate from system yanks and offer a customizable shortcut to choose what to paste within any app.

What is the current workflow expected for a PasteBar user if they want to paste, say, the 4-th most recent yank in a Safari window? Should they leave Safari, open PasteBar, open the recent history items menu, choose the one they want, and then navigate back to Safari and paste it? Or am I missing anything?

**Comment by kurdin** [comment]
> As a side note, I'd like to understand the reasoning about this app being released without this requested feature. Every clipboard manager app I've ever used (like jumpcut/flycut or maccy) is bidirectional

Honestly, I don't use PasteBar mainly as a clipboard manager. It's convenient to have unlimited history of copy/paste, but I use it mainly for quick and easy access to small pieces of important information (text, code, or images) that I frequently use or copy/paste. I copy something I need, then I create saved clips for information about my favorite music artists, my WiFi networks, kids' accounts or school contacts and teachers, API keys, my projects' contacts/marketing, important work and personal links, etc. I can organize saved clips with boards and tabs.

> What is the current workflow expected for a PasteBar user if they want to paste, 
I use the PasteBar menu for direct paste from history or clips. I always know where to find it without remembering an OS's shortcuts; there are already many of them in the system. So, using a menu, I can always find and visually see what I am about to paste to another app.

![image](https://github.com/user-attachments/assets/4498fc84-8c91-45d7-b81b-aee467a692e8)

I do understand that many people prefer to use OS shortcuts and keyboard-only access to clipboard manager history to paste from. I'll look into this in time. Thank you for your feedback.

**Comment by kurdin** [comment]
Added 2 options for customizing global OS Hotkeys 

- Show / hide main app window
- Show / hide quick paste window (still working on it) 

<img width="610" alt="image" src="https://github.com/user-attachments/assets/cf1dbd60-461d-4238-89b8-7f216296a757">

**Comment by kurdin** [comment]
New version has been released with PasteBar Quick Paste window using global hotkey.

https://github.com/PasteBar/PasteBarApp/releases/tag/v0.6.0

https://github.com/user-attachments/assets/c28ec873-4282-49c0-97a8-462742ce5b3f

**Comment by Stabledog** [comment]
Is there a global shortcut to bring PasteBar up?  It seems like the issue was resolved, but I cannot find information about the keyboard shortcut (or how to configure it?)
